sdk/identity/azure-identity/azure/identity/_credentials/browser.py

```diff
@@ -84,6 +88,49 @@ def _request_token(self, *scopes: str, **kwargs: Any) -> Dict:
         scopes = list(scopes)  # type: ignore
         claims = kwargs.get("claims")
         app = self._get_app(**kwargs)
+        if self._client_credential:
```

Comment:
nit: maybe more apt to check `if isinstance(app, msal.ConfidentialClientApplication):` since from the perspective of this method, that's the more direct condition for branching.